Samples/BackgroundBlur/BackgroundBlur/SegmentModel.cpp

```diff
@@ -0,0 +1,393 @@
+	LearningModelSession foregroundSession = CreateLearningModelSession(GetBackground(1, 3, m_imageHeightInPixels, m_imageWidthInPixels));
+	// Enable tensorcpusync for the last evaluate so can extract and give back to buffer
+	// Will remove once can just pass along d3d reources back to MFT
+	auto foregroundBinding = Evaluate(foregroundSession, std::vector<ITensor*>{&tensorizedImg, &rawLabels}, & foreground, true);
+	
+	UINT32 outCapacity = 0;
+	if (m_useGPU)
+	{
+		// v1: just get the reference- should fail
+		auto reference = foreground.as<TensorUInt8Bit>().CreateReference().data();
+
+		// v3: get from a d3dresource
+		/*ID3D12Resource* res = NULL;
+		HRESULT hr = foreground.as<ITensorNative>()->GetD3D12Resource(&res);
+		UINT DstRowPitch = 0, DstDepthPitch = 0, SrcSubresource = 0;
+		hr = res->ReadFromSubresource((void*)*pDest, DstRowPitch, DstDepthPitch, SrcSubresource, NULL);*/
+		return;
+	}
+	else 
+	{
+		auto reference = foreground.as<TensorUInt8Bit>().CreateReference().data();
+		CopyMemory(*pDest, reference, cbImageSize);
+	}
+}
+
+void SegmentModel::RunTest(const BYTE** pSrc, BYTE** pDest, DWORD cbImageSize) 
+{
+	// Right now the single input type I allow is topdown, this should work
+	winrt::array_view<const byte> source{*pSrc, *pSrc + cbImageSize}; // TODO: Does this work when topdown vs. bottomup
+	std::vector<int64_t> shape = { 1, m_imageHeightInPixels, m_imageWidthInPixels, 4 };
+
+	ITensor inputRawTensor = TensorUInt8Bit::CreateFromArray(std::vector<int64_t>{1, cbImageSize}, source);
+	ITensor outputTensor = TensorUInt8Bit::Create(shape);
+	auto binding = Evaluate(m_sess, { &inputRawTensor }, &outputTensor, true);
+
+	UINT32 outCapacity = 0;
+	if (m_useGPU)
+	{
+		// v1: just get the reference- should fail
+		auto reference = outputTensor.as<TensorUInt8Bit>().CreateReference().data();
+		CopyMemory(*pDest, reference, cbImageSize);
+
+		// v3: get from a d3dresource
+		/*ID3D12Resource* res = NULL;
+		HRESULT hr = outputTensor.as<ITensorNative>()->GetD3D12Resource(&res);
+		UINT DstRowPitch = 0, DstDepthPitch = 0, SrcSubresource = 0;
+		hr = res->ReadFromSubresource((void*)*pDest, DstRowPitch, DstDepthPitch, SrcSubresource, NULL);*/
+		return;
+	}
+}
+
+void SegmentModel::SetImageSize(UINT32 w, UINT32 h) 
+{
+	m_imageWidthInPixels = w;
+	m_imageHeightInPixels = h;
+}
+
+LearningModel SegmentModel::Argmax(long axis, long h, long w)
+{
+	auto builder = LearningModelBuilder::Create(12)
+		.Inputs().Add(LearningModelBuilder::CreateTensorFeatureDescriptor(L"Data", TensorKind::Float, array_view<int64_t const>{ -1, -1, h, w })) // Different input type? 
+		.Outputs().Add(LearningModelBuilder::CreateTensorFeatureDescriptor(L"Output", TensorKind::Float, array_view<int64_t const>{ -1, -1, h, w })) // Output of int64? 
+		.Operators().Add( LearningModelOperator(L"ArgMax")
+			.SetInput(L"data", L"Data")
+			.SetAttribute(L"keepdims", TensorInt64Bit::CreateFromArray(std::vector<int64_t>{1}, std::vector<int64_t>{ 1 }))
```

Comment:
Nitpick: There's a bunch of places with temporary std::vector. array_view accepts std::array or an initializer_list too, which are much easier and faster to build than a temporary vector that gets thrown away as soon as the call returns.

Reply:
So the parameter type here is winrt::param::iterable, so if I'm reading [Passing parameters into the ABI boundary](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/pass-parms-to-abi#iterable-parameters) I can pass std::vector and not std::array. Are there any other alternatives that are more effective? If it helps, I can move creating each of these learningmodelsessions out of the RunTest method like I was planning to do once I consolidate the models for pre- and post-processing into a single mega model.

Reply:
I was looking at the second parameter, which is array_view and can take std::array or std::initializer_list. But from the docs IIterable can also take std::initializer_list, no?

Reply:
Oh I see, yes you're right.